alica_engine/include/engine/blackboard/KeyMapping.h

```diff
@@ -11,12 +12,13 @@ class Blackboard;
 class KeyMapping
 {
 public:
-    using KeyMappingList = std::vector<std::pair<std::string, std::string>>;
+    // contains tuple of parentKey, childKey and value
+    using KeyMappingList = std::vector<std::tuple<std::string, std::string, std::string>>;
     KeyMapping() = default;
     virtual ~KeyMapping() = default;
     const KeyMappingList& getInputMapping() const;
     const KeyMappingList& getOutputMapping() const;
-    void addInputMapping(const std::string& parentKey, const std::string& childKey);
+    void addInputMapping(const std::string& parentKey, const std::string& childKey, const std::string& value = "");
```

Comment:
again, api should reflect the intention i.e.
`addInputMapping(const std::variant<std::string, std::string>& parentKeyOrConstant, const std::string& childKey)`